pkg/middleware/jwt_session.go

```diff
@@ -81,14 +81,18 @@ func (j *jwtSessionLoader) getJwtSession(req *http.Request) (*sessionsapi.Sessio
 	}
 
 	for _, loader := range j.sessionLoaders {
-		bearerToken, err := loader.Verifier.Verify(req.Context(), rawBearerToken)
-		if err == nil {
+		bearerToken, verifyErr := loader.Verifier.Verify(req.Context(), rawBearerToken)
+		if verifyErr == nil {
 			// The token was verified, convert it to a session
 			return loader.TokenToSession(req.Context(), rawBearerToken, bearerToken)
 		}
+		err = verifyErr
 	}
 
-	return nil, fmt.Errorf("unable to verify jwt token: %q", req.Header.Get("Authorization"))
+	return nil, fmt.Errorf("unable to verify jwt token: %q. Last of %d errors: %v",
+		req.Header.Get("Authorization"),
+		len(j.sessionLoaders),
+		err)
```

Comment:
As I understand this, it will only reflect errors from the last configured loader? Ideally we would present an aggregate of the errors if no loader is able to validate the token

Reply:
This might be the opportunity to cleanup how we do error handling in the `jwt_session` middleware chain.  We get a lot of questions when we print an error from a failure early in the chain only for an OIDC verifier later in the chain to succeed.

Feels like an all or nothing with error logging is ideal.  All errors if nothing in the chain succeeds.  None of the errors if any succeed.